R/simplify_network.R

```diff
@@ -12,20 +12,31 @@ simplify_network = function(rnet_y, parameters){
     rnet_x_buffer = sf::st_sf(geometry = single_rnet_x_buffer)
     rnet_x_buffer = sf::st_make_valid(rnet_x_buffer)
   } else {
-    # URL for the original route network
-    url_rnet_x = "https://github.com/nptscot/networkmerge/releases/download/v0.1/OS_Scotland_Network.geojson"
-    f_rnet_x = basename(url_rnet_x)
-    if (!file.exists(f_rnet_x)) {
-      download.file(url_rnet_x, f_rnet_x, method = "libcurl")
-    }
-    if (file.exists(f_rnet_x) && file.size(f_rnet_x) > 0) {
-      rnet_x = geojsonsf::geojson_sf(f_rnet_x)
+    # Build file path and URL based on parameters$region
+    if (!is.null(parameters$region)) {
+      region_snake_case = snakecase::to_snake_case(parameters$region)
+      base_name = paste("OS_Scotland_Network", region_snake_case, sep = "_")
     } else {
-      stop("File download failed or file is empty for rnet_x")
+      base_name = "OS_Scotland_Network"
+    }
+    
+    file_path = paste0("inputdata/", base_name, ".geojson")
+    url_rnet_x = paste0("https://github.com/nptscot/inputdata/releases/download/OS_network/", base_name, ".geojson")
```

Comment:
This looks good but I think instead of trying to download the URL you could just check if it exists in the `inputdata` folder and if not emit a message. I got this error:

```
✖ error target simplified_network
▶ end pipeline [2.931 minutes]
Warning message:
15 targets produced warnings. Run targets::tar_meta(fields = warnings, complete_only = TRUE) for the messages. 
Error:
! Error running targets::tar_make()
  Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
  Debugging guide: https://books.ropensci.org/targets/debugging.html
  How to ask for help: https://books.ropensci.org/targets/help.html
  Last error: cannot open URL 'https://github.com/nptscot/inputdata/releases/download/OS_network/OS_Scotland_Network_glasgow_and_strathclyde.geojson'
```